v2/tools/generator/internal/codegen/pipeline/apply_export_filters.go

```diff
@@ -32,47 +31,69 @@ func filterTypes(
 	configuration *config.Configuration,
 	state *State) (*State, error) {
 
-	newDefinitions := make(astmodel.Types)
-
-	filterer := configuration.BuildExportFilterer(state.types)
+	defs := make(astmodel.Types)
+	types := state.Types()
 	renames := make(map[astmodel.TypeName]astmodel.TypeName)
-
-	for _, def := range state.types {
+	for _, def := range astmodel.FindResourceTypes(types) {
 		defName := def.Name()
-		shouldExport, newName, reason := filterer(defName)
-
-		switch shouldExport {
-		case config.Skip:
-			klog.V(3).Infof("Skipping %s because %s", defName, reason)
 
-		case config.Export:
-			if reason == "" {
-				klog.V(3).Infof("Exporting %s", defName)
-			} else {
-				klog.V(2).Infof("Exporting %s because %s", defName, reason)
+		export, err := configuration.ObjectModelConfiguration.LookupExport(defName)
+		if err != nil {
+			if config.IsNotConfiguredError(err) {
+				export = false
 			}
```

Comment:
There is no path in this if statement that deals with an `err` that is _not_ a `config.NotConfiguredError`. That seems like an oversight, as an error would be silently ignored then?

Reply:
Yes, it's an oversight. At the moment the only possible error is a `NotConfiguredError`, but that might change in future. Thanks.